emcc.py

```diff
@@ -1854,6 +1838,26 @@ def include_and_export(name):
       # enables the --post-emscripten pass
       shared.Settings.GLOBAL_BASE = 1024
 
+    if shared.Settings.RELOCATABLE:
+      shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$reportUndefinedSymbols', '$relocateExports', '$GOTHandler']
+      shared.Settings.EXTRA_WASM_IMPORTS += [
+        # tell the memory segments where to place themselves
+        ('__memory_base', str(shared.Settings.GLOBAL_BASE)),
+        ('__indirect_function_table', 'wasmTable'),
+        # the backend reserves slot 0 for the NULL function pointer
+        ('__table_base', '1'),
```

Comment:
I think these are the only three that are need the key, value pairs.   So if leave these three as way were before you can move this block back up to its original position and just add `__stack_pointer` here... this making this patch much smaller.  We can than look at ways to address these three outliers separately.

Reply:
I believe as part of my change to way MAIN_MODULE works all three of these may well be going away soon anyway.